conduit/plugins/exporters/filewriter/util_test.go

```diff
@@ -1,13 +1,109 @@
 package filewriter
 
 import (
-	"io/ioutil"
+	"os"
 	"path"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 )
 
+func TestParseFilenameFormat(t *testing.T) {
+	testCases := []struct {
+		name        string
+		format      string
+		gzip        bool
+		blockFormat EncodingFormat
+		err         string
+	}{
+		{
+			name:        "messagepack vanilla",
+			format:      "%d_block.msgp",
+			gzip:        false,
+			blockFormat: MessagepackFormat,
+			err:         "",
+		},
+		{
+			name:        "messagepack gzip",
+			format:      "%d_block.msgp.gz",
+			gzip:        true,
+			blockFormat: MessagepackFormat,
+			err:         "",
+		},
+		{
+			name:        "json vanilla",
+			format:      "%d_block.json",
+			gzip:        false,
+			blockFormat: JSONFormat,
+			err:         "",
+		},
+		{
+			name:        "json gzip",
+			format:      "%d_block.json.gz",
+			gzip:        true,
+			blockFormat: JSONFormat,
+			err:         "",
+		},
+		{
+			name:        "messagepack vanilla 2",
+			format:      "%[1]d_block round%[1]d.msgp",
+			gzip:        false,
+			blockFormat: MessagepackFormat,
+			err:         "",
+		},
+		{
+			name:        "messagepack gzip 2",
+			format:      "%[1]d_block round%[1]d.msgp.gz",
+			gzip:        true,
+			blockFormat: MessagepackFormat,
+			err:         "",
+		},
+		{
+			name:        "json vanilla 2",
+			format:      "%[1]d_block round%[1]d.json",
+			gzip:        false,
+			blockFormat: JSONFormat,
+			err:         "",
+		},
+		{
+			name:        "json gzip 2",
+			format:      "%[1]d_block round%[1]d.json.gz",
+			gzip:        true,
+			blockFormat: JSONFormat,
+			err:         "",
+		},
+		{
+			name:        "invalid - gzip",
+			format:      "%d_block.msgp.gzip",
+			gzip:        false,
+			blockFormat: UnrecognizedFormat,
+			err:         "unrecognized export format",
+		},
+		{
+			name:        "invalid - no extension",
+			format:      "%d_block",
+			gzip:        false,
+			blockFormat: UnrecognizedFormat,
+			err:         "unrecognized export format",
+		},
```

Comment:
add another case for format without the %d specifier?

Reply:
I originally did, but unfortunately it doesn't fail, just gives a really gnarly file name that likely wouldn't get saved:

https://go.dev/play/p/S4QyqmUJVp6

I could add some explicit checks, but it's not obvious to me exactly what they should check for.

Reply:
that's fine as long as we document the expected filename pattern.